oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/TraceQueryEsDAO.java

```diff
@@ -162,7 +163,11 @@ public TraceBrief queryBasicTraces(long startSecondTB,
         sourceBuilder.query(QueryBuilders.termQuery(SegmentRecord.TRACE_ID, traceId));
         sourceBuilder.size(segmentQueryMaxSize);
 
-        SearchResponse response = getClient().search(SegmentRecord.INDEX_NAME, sourceBuilder);
+        String traceIdThirdPart = traceId.split("\\.")[2];
+        long timeBucket = TimeBucket.getRecordTimeBucket(Long.valueOf(traceIdThirdPart) / 10000);
+
+        SearchResponse response = getClient().search(
+            new TimeRangeIndexNameMaker(SegmentRecord.INDEX_NAME, timeBucket, timeBucket), sourceBuilder);
```

Comment:
I am afraid this is not working. We don't have this timestamp-based trace-id rule anymore.

Reply:
Take a look at other agent implementations, such as LUA.

Reply:
> Take a look at other agent implementations, such as LUA.

Ok let me take a look.

Reply:
@wu-sheng I modified some content, can you help me see if this is feasible?

Reply:
From my personal understanding, we don't have the format rule of traceid anymore. Any string is acceptable, and even it is combined with 3 parts, doesn't mean there is a timestamp included.